### DISCUSSION TITLE
side_ch.h slv_reg limited access

### DISCUSSION THREAD

**Discussion by abragadabra7** [open-sdr/openwifi#309] [Q&A] [answered]
hello,
i am learning openwifi project and i came across side_ch component.
side_ch axi-lite controlled by PS, according to side_ch.h it doesnt have all the addresses of all the slv_reg(s) , does it mean the code is not complete or there is no use of the other 9 registers(19,21,22,26-31) in side_ch_axi?

**Comment by JiaoXianjun** [comment]
We commented out some unused registers to save resource. You can uncomment them if you want.

**Reply by abragadabra7** [reply]
thank you for the fast replay,
I am talking about side_ch.h, it has the following defines:
```
#define SIDE_CH_REG_MULTI_RST_ADDR                 (0*4)
#define SIDE_CH_REG_CONFIG_ADDR                    (1*4)
#define SIDE_CH_REG_NUM_DMA_SYMBOL_ADDR            (2*4) //low 16bit to PS; high 16bit to PL
#define SIDE_CH_REG_IQ_CAPTURE_ADDR                (3*4)
#define SIDE_CH_REG_NUM_EQ_ADDR                    (4*4)
#define SIDE_CH_REG_FC_TARGET_ADDR                 (5*4)
#define SIDE_CH_REG_ADDR1_TARGET_ADDR              (6*4)
#define SIDE_CH_REG_ADDR2_TARGET_ADDR              (7*4)
#define SIDE_CH_REG_IQ_TRIGGER_ADDR                (8*4)
#define SIDE_CH_REG_RSSI_TH_ADDR                   (9*4)
#define SIDE_CH_REG_GAIN_TH_ADDR                   (10*4)
#define SIDE_CH_REG_PRE_TRIGGER_LEN_ADDR           (11*4)
#define SIDE_CH_REG_IQ_LEN_ADDR                    (12*4)

#define SIDE_CH_REG_M_AXIS_DATA_COUNT_ADDR         (20*4)
```
but side_ch.v and side_ch_s_axi.v have 19,21,22,26-31 used.
 does it mean PS/driver parts doesnt have access to these registers?

**Reply by JiaoXianjun** [reply, marked as answer]
yes
